### PR TITLE
Add some more tests and remove duplicates

### PR DESCRIPTION
**Pull request by millerdev**
As added in https://github.com/dimagi/commcare-hq/pull/17000

@ctsims

### CODE REVIEW

File: src/test/resources/case_relationship_tests.json [hunk 1]
```diff
@@ -636,40 +636,6 @@
...
-                "b"
-            ]
-        ],
-        "closed": [
-            "b"
-        ],
-        "outcome": []
-    },
-    {
-        "name": "closed_extension",
```
Comment (millerdev) [inline src/test/resources/case_relationship_tests.json]: `AssertionError: duplicate tests: closed_extension, Closed_Extension (owned:0 closed:1 outcome:0 extensions:1,0)`

duplicate: https://github.com/dimagi/commcare-core/blob/0faea8bffa52fc9ae7d386d9bfc357a10d175fce/src/test/resources/case_relationship_tests.json#L406-L421

File: src/test/resources/case_relationship_tests.json [hunk 2]
```diff
@@ -636,40 +636,6 @@
         "outcome": []
     },
     {
-        "name": "owned_closed_open_extension",
```
Comment (millerdev) [inline src/test/resources/case_relationship_tests.json]: `AssertionError: duplicate tests: owned_closed_open_extension, Delegate_Closure (owned:0_closed:1_extensions:0,1)`

duplicate: https://github.com/dimagi/commcare-core/blob/0faea8bffa52fc9ae7d386d9bfc357a10d175fce/src/test/resources/case_relationship_tests.json#L548-L561
